src/llama_recipes/utils/train_utils.py

```diff
@@ -179,17 +191,14 @@ def train(model, train_dataloader,eval_dataloader, tokenizer, optimizer, lr_sche
             checkpoint_times.append(checkpoint_end_time)
             if eval_epoch_loss < best_val_loss:
                 best_val_loss = eval_epoch_loss
-                if train_config.enable_fsdp:
-                    if rank==0:
-                        print(f"best eval loss on epoch {epoch+1} is {best_val_loss}")
-                else:
+                if track_metrics:
                     print(f"best eval loss on epoch {epoch+1} is {best_val_loss}")
+            if tracker is not None and track_metrics:
+                tracker.track(best_val_loss , name='best_val_loss', stage='validation')
+                tracker.track(eval_epoch_loss , name='eval_epoch_loss', stage='validation')
             val_loss.append(best_val_loss)
             val_prep.append(eval_ppl)
-        if train_config.enable_fsdp:
-            if rank==0:
-                print(f"Epoch {epoch+1}: train_perplexity={train_perplexity:.4f}, train_epoch_loss={train_epoch_loss:.4f}, epoch time {epoch_end_time}s")
-        else:
+        if track_metrics:
```

Comment:
it looks like this change is removing the logging of epoch loss and time in the fsdp case?

Reply:
The original code checks if fsdp is enabled and if so prints only on rank 0. if fsdp is not enabled it prints anyway.

`track_metrics` is used as a short cut to check this condition. It is set as `not (train_config.enable_fsdp and rank != 0)` to mimic this condition and remove a duplicate print statement.

Reply:
ah maybe I misread the condition the first time